Microsoft.Toolkit.Uwp.UI.Controls.Input/TokenizingTextBox/TokenizingTextBoxAutomationPeer.cs

```diff
@@ -0,0 +1,131 @@
+        /// <returns>**true** if the value is read-only; **false** if it can be modified.</returns>
+        public bool IsReadOnly => !this.OwningTokenizingTextBox.IsEnabled;
+
+        /// <summary>Gets the value of the control.</summary>
+        /// <returns>The value of the control.</returns>
+        public string Value => this.OwningTokenizingTextBox.Text;
+
+        private TokenizingTextBox OwningTokenizingTextBox
+        {
+            get
+            {
+                return Owner as TokenizingTextBox;
+            }
+        }
+
+        /// <summary>Sets the value of a control.</summary>
+        /// <param name="value">The value to set. The provider is responsible for converting the value to the appropriate data type.</param>
+        /// <exception cref="T:Windows.UI.Xaml.Automation.ElementNotEnabledException">Thrown if the control is in a read-only state.</exception>
+        public void SetValue(string value)
+        {
+            if (IsReadOnly)
+            {
+                throw new ElementNotEnabledException($"Could not set the value of the {nameof(TokenizingTextBox)} ");
+            }
+
+            this.OwningTokenizingTextBox.Text = value;
+        }
+
+        /// <summary>
+        /// Called by GetClassName that gets a human readable name that, in addition to AutomationControlType,
+        /// differentiates the control represented by this AutomationPeer.
+        /// </summary>
+        /// <returns>The string that contains the name.</returns>
+        protected override string GetClassNameCore()
+        {
+            return Owner.GetType().Name;
+        }
+
+        /// <summary>
+        /// Called by GetName.
+        /// </summary>
+        /// <returns>
+        /// Returns the first of these that is not null or empty:
+        /// - Value returned by the base implementation
+        /// - Name of the owning TokenizingTextBox
+        /// - TokenizingTextBox class name
+        /// </returns>
+        protected override string GetNameCore()
+        {
+            string name = this.OwningTokenizingTextBox.Name;
+            if (!string.IsNullOrWhiteSpace(name))
+            {
+                return name;
+            }
+
+            name = AutomationProperties.GetName(this.OwningTokenizingTextBox);
+            return !string.IsNullOrWhiteSpace(name) ? name : base.GetNameCore();
+        }
+
+        /// <summary>
+        /// Gets the control pattern that is associated with the specified Windows.UI.Xaml.Automation.Peers.PatternInterface.
+        /// </summary>
+        /// <param name="patternInterface">A value from the Windows.UI.Xaml.Automation.Peers.PatternInterface enumeration.</param>
+        /// <returns>The object that supports the specified pattern, or null if unsupported.</returns>
+        protected override object GetPatternCore(PatternInterface patternInterface)
+        {
+            return patternInterface switch
+            {
+                PatternInterface.Value => this,
```

Comment:
What about ItemContainer? After all we are a list of Tokens and according to docs, controls that are ItemsControl are also IItemContainerProviers. Or is that interface covered by calling base?

Reply:
This should be handled by the `ListViewBaseAutomationPeer` as it inherits from the `ItemsControlAutomationPeer` that should handle this.

Reply:
If ListViewBaseAutomationPeer handles this, we are good here :)